lona/html/data_binding/inputs.py

```diff
@@ -1,101 +1,35 @@
 
 DEFAULT_INPUT_DELAY = 300
 
 
-class TextInput(Widget):
-    NODE_CLASS = InputNode
-
-    ID_LIST = []
-    CLASS_LIST = []
-    STYLE = {}
+class TextInput(Node):
+    TAG_NAME = 'input'
+    SELF_CLOSING_TAG = True
 
     ATTRIBUTES = {
         'type': 'text',
     }
 
-    def __init__(self, value=None, disabled=False, bubble_up=False,
-                 input_delay=DEFAULT_INPUT_DELAY, **input_kwargs):
-
-        self.bubble_up = bubble_up
-
-        self.input_node = self.gen_input_node(**input_kwargs)
+    def __init__(self, *args, value=None, disabled=False, bubble_up=False,
+                 input_delay=DEFAULT_INPUT_DELAY, **kwargs):
 
-        self.input_node.events.add(CHANGE(input_delay))
+        super().__init__(*args, **kwargs)
 
-        self.nodes = [
-            self.input_node,
-        ]
+        self.events.add(CHANGE(input_delay))
 
+        self.bubble_up = bubble_up
         self.disabled = disabled
 
         if value is not None:
             self.value = value
 
-    def gen_node_args(self, **kwargs):
-        id_list = copy(self.ID_LIST)
-        class_list = copy(self.CLASS_LIST)
-        style = copy(self.STYLE)
-        attributes = copy(self.ATTRIBUTES)
-        misc_kwargs = {}
-
-        for name, value in kwargs.copy().items():
-
-            # remove underscores from attributes
-            # this makes kwargs like '_class' possible to prevent clashes
-            # with python grammar
-            clean_name = name
-
-            if '_' in clean_name:
-                clean_name = clean_name.replace('_', '-')
-
-                if clean_name.startswith('-'):
-                    clean_name = clean_name[1:]
-
-            if clean_name == 'id':
-                value = kwargs.pop(name)
-
-                if isinstance(value, str):
-                    value = value.split(' ')
-
-                id_list.extend(value)
-
-            elif clean_name == 'class':
-                value = kwargs.pop(name)
-
-                if isinstance(value, str):
-                    value = value.split(' ')
-
-                class_list.extend(value)
-
-            elif clean_name == 'style':
-                style.update(kwargs.pop(name))
-
-            elif clean_name == 'attributes':
-                attributes.update(kwargs.pop(name))
-
-            else:
-                misc_kwargs[name] = value
-
-        return {
-            'id': id_list,
-            'class': class_list,
-            'style': style,
-            **attributes,
-            **misc_kwargs,
-        }
-
-    def gen_input_node(self, **kwargs):
-        return self.NODE_CLASS(**self.gen_node_args(**kwargs))
-
     def handle_input_event(self, input_event):
         if input_event.name != 'change':
             return input_event
 
-        self.input_node.attributes.__setitem__(
+        self.attributes.__setitem__(
             'value',
             input_event.data,
             issuer=(input_event.connection, input_event.window_id),
```

Comment:
nit: Let's add explaining comment (looks like some hack)

Reply:
I added the commit message of the original commit. Hm it could look like a hack because i use ``__setitem__`` directly. Maybe we should move that functionality to another function. Maybe not in this pr though.